src/MudBlazor/Extensions/DataGridExtensions.cs

```diff
@@ -10,6 +10,18 @@ namespace MudBlazor
 #nullable enable
     public static class DataGridExtensions
     {
+        public static IEnumerable<T> ApplyFilters<T>(this IEnumerable<T> source,
+            GridState<T> state, FilterOptions? filterOptions = null)
+            => ApplyFilters(source, state.FilterDefinitions, filterOptions);
+
+        public static IEnumerable<T> ApplyFilters<T>(this IEnumerable<T> source,
+            ICollection<IFilterDefinition<T>> filterDefinitions, FilterOptions? filterOptions)
+        {
+            IEnumerable<T> sourceArray = source as T[] ?? source.ToArray();
+            return filterDefinitions.Aggregate(
+                sourceArray, (current, filterDefinition) => current.Where(filterDefinition.GenerateFilterFunction(filterOptions)));
+        }
+
```

Comment:
I would rename `ApplyFilters` to simply `Where` and simplify (no need to convert to an array here, aggregate works on an `IEnumerable<T>` without multiple enumerations). 

```suggestion
        public static IEnumerable<T> Where<T>(this IEnumerable<T> source, GridState<T> state, FilterOptions? filterOptions = null)
            => Where(source, state.FilterDefinitions, filterOptions);

        public static IEnumerable<T> Where<T>(this IEnumerable<T> source, ICollection<IFilterDefinition<T>> filterDefinitions, FilterOptions? filterOptions)
            => filterDefinitions.Aggregate(source, (current, filter) => current.Where(filter.GenerateFilterFunction(filterOptions)));
```